js/app.js

```diff
@@ -0,0 +1,517 @@
+"use strict"
+
+// Basic exercises
+for (let i =0; i < 5; i++) {
+  console.log('hello world')
+}
+
+let person = {
+  name: 'josh',
+  age: 18
+};
+
+person.name = 'Bosh';
+console.log(person);
+person['name']='Mary'
+console.log(person);
+
+let selectedColors = ['red', 'green', 'blue'];
+selectedColors [3] = 'purple'
+console.log(selectedColors);
+console.log(selectedColors.length);
+
+ function greet (name, lastname) {
+   console.log('hello ' + name + '' + lastname);
+ }
+
+greet('Victor', 'Oshimen');
+greet('Mary', 'Jane');
+
+function square(number) {
+  return number * number;
+}
+console.log(square(5));
+
+
+function increase(number) {
+  number++;
+  return number;
+}
+
+console.log(increase(5));
+
+
+
+// Homework
+const thingsInHome = ["mint", "basil", "cactus", "table", "wooden spoon", "bread"];
+const thingsInGarden = ["apple", "trees", "stairs", "plum", "wooden bench"];
+
+// -- edit below --
+const madeOutOfWood = [thingsInHome[3], thingsInGarden[4]];
+const edibles = [thingsInHome[0], thingsInHome[1], thingsInHome[5], thingsInGarden[0], thingsInGarden[3]];
+// -- edit above --
+
+console.log(
+  `made out of wood: ${JSON.stringify(madeOutOfWood)}\n edibles: ${JSON.stringify(edibles)}`
+);
+
+
+// 2 for loop
+function getSmallestNumber(numbers) {
+  let smallestNumber = numbers[0]
+  for (let i = 0; i < numbers.length; ++i) {
+    if (smallestNumber > numbers[i]) {
+     smallestNumber = numbers[i];
+    }
+  }
+  return smallestNumber;
+}
+
+console.log(getSmallestNumber([2, -5, 10, 1, 4])); // -5
+getSmallestNumber([200, 25, 4, 123, 87]); // 4
+
+// 2.5 sort function
+
+function giveSmallestNumber(numbers) {
+  return numbers.sort((a, b) => a - b)[0];
+}
+
+console.log(giveSmallestNumber([2, -5, 10, 1, 4])); // -5
+console.log(giveSmallestNumber([200, 25, 4, 123, 87])); // 4
+
+// 3
+console.log(getSquaredNumbers([1, 2, 3, 4, 5])); // [1, 4, 9, 16, 25]
+console.log(getSquaredNumbers([6, 7, 8, 9, 10])); // [36, 49, 64, 81, 100]
+
+function getSquaredNumbers(numbers) {
+  let squaredArray = []
+  for (let i = 0; i < numbers.length; i++) {
+    let square = numbers[i] * numbers[i];
+    squaredArray.push(square);
+  }
+  return squaredArray;
+}
+
+// function
+// make new array
+// lets multiply the array by itself
+// push it in the array
+// return the squaredArray
+
+const numbers = [1, 2, 3];
+const squaredNumbers = getSquaredNumbers(numbers);
+
+console.log(squaredNumbers); // [1, 4, 9]
+console.log(numbers); // [1, 2, 3]
+console.log(numbers !== squaredNumbers); // true
+
+// 4
+function getReversedString(string) {
+  return string.split('').reverse().join('').replace(/\s+/g, '');
```

Comment:
Why do you need this `.replace(/\s+/g, '')` bit?

Reply:
https://flaviocopes.com/how-to-replace-whitespace-javascript/
it is to replace the spaces between letters